src/frontends/tensorflow/src/helper_transforms/gru_block_cell_replacer.cpp

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "helper_transforms/gru_block_cell_replacer.hpp"
+
+#include <memory>
+#include <vector>
+
+#include "helper_ops/gru_block_cell.hpp"
+#include "ngraph/rt_info.hpp"
+#include "openvino/opsets/opset9.hpp"
+#include "openvino/pass/pattern/matcher.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+#include "utils.hpp"
+
+using namespace std;
+using namespace ov::pass;
+using namespace ov::opset9;
+using namespace ov::frontend::tensorflow;
+
+pass::GRUBlockCellReplacer::GRUBlockCellReplacer() {
+    auto gru_block_cell = pattern::wrap_type<GRUBlockCell>();
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        NodeRegistry rg;
+
+        auto gru_block_cell_node = std::dynamic_pointer_cast<GRUBlockCell>(m.get_match_root());
+        if (!gru_block_cell_node) {
+            return false;
+        }
+
+        // this transformation expects only one output (the forth output) - hidden state
+        // that is only output supported by OpenVINO GRUCell
+        std::vector<int> restricted_output_indices = {0, 1, 2};
+        for (size_t output_ind : restricted_output_indices) {
+            if (gru_block_cell_node->output(output_ind).get_target_inputs().size() > 0) {
+                return false;
+            }
+        }
+
+        // currently, OpenVINO support only static hidden_size
+        auto m_hidden_size = gru_block_cell_node->get_hidden_size();
+        if (m_hidden_size.is_dynamic()) {
+            return false;
+        }
+
+        auto x = gru_block_cell_node->input_value(0);
+        auto h_prev = gru_block_cell_node->input_value(1);
+        auto w_ru = gru_block_cell_node->input_value(2);
+        auto w_c = gru_block_cell_node->input_value(3);
+        auto b_ru = gru_block_cell_node->input_value(4);
+        auto b_c = gru_block_cell_node->input_value(5);
+
+        // retrive input_size and hidden_size
+        auto x_shape = rg.make<ShapeOf>(x, element::i64);
+        auto ss_start = rg.make<Constant>(element::i64, Shape{1}, 1);
+        auto ss_end = rg.make<Constant>(element::i64, Shape{1}, 2);
+        auto ss_step = rg.make<Constant>(element::i64, Shape{1}, 1);
+        auto input_size = rg.make<Slice>(x_shape, ss_start, ss_end, ss_step);
+        auto h_prev_shape = rg.make<ShapeOf>(h_prev, element::i64);
+        auto hidden_size = rg.make<Slice>(h_prev_shape, ss_start, ss_end, ss_step);
```

Comment:
could you check case where hidden_size and input_size are dynamic in input shape (h_prev_shape)?
please add such a test